lib/oregon_digital/controlled_vocabularies/scientific.rb

```diff
@@ -17,45 +17,30 @@ def self.all_endpoints
       end
 
       def fetch(*_args, &_block)
-        case self.class.query_to_vocabulary(rdf_subject.to_s)
+        vocabulary = self.class.query_to_vocabulary(rdf_subject.to_s)
```

Comment:
I'm sure you're headed this way, but everything in this file looks generic enough. Push it up to resource and share it with subject